sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java

```diff
@@ -2698,6 +2698,38 @@ public VirtualMachineImpl withExistingVirtualMachineScaleSet(VirtualMachineScale
         return this;
     }
 
+    @Override
+    public VirtualMachineImpl withOSDisk(String diskId) {
+        if (diskId == null) {
+            return this;
+        }
+        return withOSDisk(diskId, ResourceUtils.nameFromResourceId(diskId));
+    }
+
+    @Override
+    public VirtualMachineImpl withOSDisk(String diskId, String diskName) {
+        if (!isManagedDiskEnabled() || this.innerModel().storageProfile().osDisk().managedDisk() == null) {
+            return this;
+        }
+        OSDisk osDisk = new OSDisk()
```

Comment:
Need to set a brand new OSDisk for VM, otherwise some remaining properties(like encryption settings) may cause update to fail.
On the other hand, since it's total replacement of the os disk, it's reasonable to replace the OSDisk as a whole.